src/Servers/Kestrel/Core/src/Internal/Infrastructure/KestrelEventSource.cs

```diff
@@ -261,6 +299,11 @@ protected override void OnEventCommand(EventCommandEventArgs command)
                 {
                     DisplayName = "Connection Queue Length"
                 };
+
+                _http2RequestQueueLengthCounter ??= new PollingCounter("http2-request-queue-length", this, () => _http2RequestQueueLength)
```

Comment:
Should HTTP/2 in the counter name? If HTTP/3 has the same architecture then a neutral name would allow the counter to be reused.

Reply:
I was looking to see if I could do the same for all HTTP versions but it might make sense to have a counter for each since we don’t have dimensions